src/NuGet.Core/NuGet.Versioning/SemanticVersionFactory.cs

```diff
@@ -95,42 +101,74 @@ internal static bool IsLetterOrDigitOrDash(char c)
             return (x >= 48 && x <= 57) || (x >= 65 && x <= 90) || (x >= 97 && x <= 122) || x == 45;
         }
 
-        internal static bool IsValid(string s, bool allowLeadingZeros)
+        internal static bool IsDigit(char c)
         {
-            return s.Split('.').All(p => IsValidPart(p, allowLeadingZeros));
+            var x = (int)c;
+
+            // "0123456789"
+            return (x >= 48 && x <= 57);
```

Comment:
I see now it could be. I thought that allowed localized numbers, but that is Char.IsNumber apparently.